src/common/cluster-store/cluster-store.ts

```diff
@@ -115,7 +116,7 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
   }
 
   addCluster(clusterOrModel: ClusterModel | Cluster): Cluster {
-    appEventBus.emit({ name: "cluster", action: "add" });
+    this.dependencies.emitAppEvent({ name: "cluster", action: "add" });
```

Comment:
I'm thinking `emitAppEvent` is too low an abstraction for this context. It could instead be `emitClusterAdd()` or maybe even `emit<ClusterEvent>("add")`. The former would replace the logical dependency with a more trackable one. Eg. we could discover listeners of `ClusterEvent` more easily (if even present in this codebase). Also it could add type safety to the payload of the event, ie. `action: "add"`.